mantle/cmd/kola/testiso.go

```diff
@@ -679,55 +679,104 @@ func awaitCompletion(ctx context.Context, inst *platform.QemuInstance, outdir st
-		for _, exp := range expected {
-			l, err := r.ReadString('\n')
+		go func() {
+			err := inst.Wait()
+			// only one Wait() gets process data, so also manually check for signal
+			plog.Debugf("qemu exited err=%v", err)
+			if err == nil && inst.Signaled() {
+				err = errors.New("process killed")
+			}
 			if err != nil {
-				if err == io.EOF {
-					// this may be from QEMU getting killed or exiting; wait a bit
-					// to give a chance for .Wait() above to feed the channel with a
-					// better error
-					time.Sleep(1 * time.Second)
-					errchan <- fmt.Errorf("Got EOF from completion channel, %s expected", exp)
-				} else {
-					errchan <- errors.Wrapf(err, "reading from completion channel")
+				errchan <- errors.Wrapf(err, "QEMU unexpectedly exited while awaiting completion")
+			}
+			time.Sleep(1 * time.Minute)
+			errchan <- fmt.Errorf("QEMU exited; timed out waiting for completion")
+		}()
+		go func() {
+			r := bufio.NewReader(qchan)
+			for _, exp := range expected {
+				l, err := r.ReadString('\n')
+				if err != nil {
+					if err == io.EOF {
+						// this may be from QEMU getting killed or exiting; wait a bit
+						// to give a chance for .Wait() above to feed the channel with a
+						// better error
+						time.Sleep(1 * time.Second)
+						errchan <- fmt.Errorf("Got EOF from completion channel, %s expected", exp)
+					} else {
+						errchan <- errors.Wrapf(err, "reading from completion channel")
+					}
+					return
 				}
-				return
+				line := strings.TrimSpace(l)
+				if line != exp {
+					errchan <- fmt.Errorf("Unexpected string from completion channel: %s expected: %s", line, exp)
+					return
+				}
+				plog.Debugf("Matched expected message %s", exp)
 			}
-			line := strings.TrimSpace(l)
-			if line != exp {
-				errchan <- fmt.Errorf("Unexpected string from completion channel: %s expected: %s", line, exp)
-				return
+			plog.Debugf("Matched all expected messages")
+			// OK!
+			errchan <- nil
+		}()
+		go func() {
+			//check for error when switching boot order
+			if booterrchan != nil {
+				if err := <-booterrchan; err != nil {
+					errchan <- err
+				}
 			}
-			plog.Debugf("Matched expected message %s", exp)
-		}
-		plog.Debugf("Matched all expected messages")
-		// OK!
-		errchan <- nil
-	}()
-	go func() {
-		//check for error when switching boot order
-		if booterrchan != nil {
-			if err := <-booterrchan; err != nil {
-				errchan <- err
+		}()
+		err := <-errchan
+		elapsed := time.Since(start)
+		if err == nil {
+			// No error so far, check the console and journal files
+			consoleFile := filepath.Join(outdir, "console.txt")
+			journalFile := filepath.Join(outdir, "journal.txt")
+			files := []string{consoleFile, journalFile}
+			for _, file := range files {
+				fileName := filepath.Base(file)
+				// Check if the file exists
+				_, err := os.Stat(file)
+				if os.IsNotExist(err) {
+					fmt.Printf("The file: %v does not exist\n", fileName)
+					continue
```

Comment:
did you encounter a situation where one of these files didn't exist? IIUC they always exist, just may be empty.

Reply:
if they will always exist I'd prefer to simplify the code.

Reply:
In my understanding as per [newQemuBuilder](https://github.com/coreos/coreos-assembler/blob/main/mantle/cmd/kola/testiso.go#L408) the `journal.txt` should be created in the directory specified by `outdir` when the `forwardJournal` func is called. However... whenever I build kola, and run the test `journal.txt` is never created.

```
Running test: iso-live-login.uefi
No badness detected in the file: console.txt
The file: journal.txt does not exist
PASS: iso-live-login.uefi (16.919s)
```
I asked previously about error injection as I thought this would trigger the journal. I successfully injected the error, but the journal.txt was never there, hence the `os.IsNotExist` logic. As of now the result always presents missing journal.txt file.